<?xml version="1.0"?>
<launch>

    <arg name="init_pose"/>

    <!-- Spawn a camera into Gazebo -->
    <node name="spawner" pkg="gazebo_ros" type="spawn_model" output="screen"
        args="-urdf -param camera1/robot_description -model camera $(arg init_pose)"/>

    <!-- Convert joint states to TF transforms for rviz, etc -->
    <node ns="camera1" name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
        respawn="false" output="screen">
        <!-- <remap from="/joint_states" to="/camera1/joint_states" /> -->
    </node>

    <!-- Indicate in the tf tree that the camera and platform's worlds coincide -->
    <node name="static_tf2_world_broadcaster" type="static_tf2_world_broadcaster.py" pkg="camera"/>

    <!-- Add a new frame to retrieve the right theorical orientation of the camera -->
    <node name="new_camera_frame" type="new_camera_frame.py" pkg="camera"/>

</launch>